binding-mri/fmod-binding/studio/eventinstance.cpp

```diff
@@ -0,0 +1,463 @@
+
+RB_METHOD(eventInstanceStart)
+{
+    RB_UNUSED_PARAM;
+
+    EventInstance *b = getPrivateData<EventInstance>(self);
+
+    FMOD_RESULT result = FMOD_Studio_EventInstance_Start(b->p);
+
+    FMOD_RESULT_SIMPLE;
+}
+
+RB_METHOD(eventInstanceStop)
+{
+    int mode;
+    rb_get_args(argc, argv, "i", &mode RB_ARG_END);
+
+    EventInstance *b = getPrivateData<EventInstance>(self);
+
+    FMOD_RESULT result = FMOD_Studio_EventInstance_Stop(b->p, (FMOD_STUDIO_STOP_MODE)mode);
+
+    FMOD_RESULT_SIMPLE;
+}
+
+RB_METHOD(eventInstanceGetTimelinePosition)
+{
+    RB_UNUSED_PARAM;
+
+    EventInstance *b = getPrivateData<EventInstance>(self);
+    int position;
+
+    FMOD_RESULT result = FMOD_Studio_EventInstance_GetTimelinePosition(b->p, &position);
+
+    FMOD_RESULT_CONVERT(position, INT2NUM);
+}
+
+RB_METHOD(eventInstanceSetTimelinePosition)
+{
+    int position;
+    rb_get_args(argc, argv, "i", &position RB_ARG_END);
+
+    EventInstance *b = getPrivateData<EventInstance>(self);
+
+    FMOD_RESULT result = FMOD_Studio_EventInstance_SetTimelinePosition(b->p, position);
+
+    FMOD_RESULT_SIMPLE;
+}
+
+RB_METHOD(eventInstanceGetChannelGroup)
+{
+    // TODO
+}
+
+RB_METHOD(eventInstanceGetMinMaxDistance)
+{
+    RB_UNUSED_PARAM;
+
+    EventInstance *b = getPrivateData<EventInstance>(self);
+    float min;
+    float max;
+
+    FMOD_RESULT result = FMOD_Studio_EventInstance_GetMinMaxDistance(
+        b->p, &min, &max);
+
+    FMOD_RESULT_BASE;
+    if (result == FMOD_OK)
+    {
+        rb_ary_push(return_ary, DBL2NUM(min));
+        rb_ary_push(return_ary, DBL2NUM(max));
+    }
+    FMOD_RESULT_RET;
+}
+
+RB_METHOD(eventInstanceRelease)
+{
+    RB_UNUSED_PARAM;
+
+    EventInstance *b = getPrivateData<EventInstance>(self);
+
+    FMOD_RESULT result = FMOD_Studio_EventInstance_Release(b->p);
+
+    FMOD_RESULT_SIMPLE;
+}
+
+RB_METHOD(eventInstanceIsVirtual)
+{
+    RB_UNUSED_PARAM;
+
+    EventInstance *b = getPrivateData<EventInstance>(self);
+    FMOD_BOOL isVirtual;
+
+    FMOD_RESULT result = FMOD_Studio_EventInstance_IsVirtual(b->p, &isVirtual);
+
+    FMOD_RESULT_CONVERT(isVirtual, BOOL2RB);
+}
+
+#define RETURN_PARAMETER                              \
+    FMOD_RESULT_BASE;                                 \
+    if (result == FMOD_OK)                            \
+    {                                                 \
+        rb_ary_push(return_ary, DBL2NUM(value));      \
+        rb_ary_push(return_ary, DBL2NUM(finalvalue)); \
+    }                                                 \
+    FMOD_RESULT_RET;
+
+RB_METHOD(eventInstanceGetParameterByName)
+{
+    char *name;
+    rb_get_args(argc, argv, "z", &name RB_ARG_END);
+
+    EventInstance *b = getPrivateData<EventInstance>(self);
+    float value, finalvalue;
+
+    FMOD_RESULT result = FMOD_Studio_EventInstance_GetParameterByName(
+        b->p, name, &value, &finalvalue);
+
+    RETURN_PARAMETER;
+}
+
+RB_METHOD(eventInstanceSetParameterByName) {
```

Comment:
I'm getting SEGFAULT when I trying to change parameter to my event instance

Reply:
![Traceback](https://i.imgur.com/nHR0M8g.png)

Reply:
Can you provide your ruby code?

Reply:
> Can you provide all of your ruby code?

Sure!

Main.rb:
```rb
begin
  $console = Graphics.fullscreen
  Graphics.frame_rate = 60
  Font.default_size = 20

  fresult, $fmod = FMOD::Studio::System.new(32, 1, 0)

  warn "FMOD Result: #{fresult} | System: #{$fmod} | Valid?: #{$fmod.is_valid}"

  fresult, $bank = $fmod.load_bank_file("Audio/Banks/Master.bank", 0)
  fresult, $bgmbank = $fmod.load_bank_file("Audio/Banks/BGMs.bank", 0)
  fresult, $ambbank = $fmod.load_bank_file("Audio/Banks/Ambients.bank", 0)
  fresult, $strbank = $fmod.load_bank_file("Audio/Banks/Master.strings.bank", 0)

  $fmod.update

  fresult, $evdesc = $fmod.get_event("event:/BGMs/BGM01")
  fresult, $evinst = $evdesc.create_instance

  $fmod.update

  # Load persistent data
  Persistent.load

  ...
end
```

Audio_FMOD.rb:
```rb
module Audio

  class << self
    def method_missing(*args)
    end
  end

end

module FMODPlayer

  def self.start
    $evinst.start
    $fmod.update
  end

  def self.stop
    $evinst.stop(0)
    $fmod.update
  end

  def self.set(param, value, ignoreseek = false)
    $evinst.set_parameter_by_name(param, value, ignoreseek)
    $fmod.update
  end

end
```

Calling `Audio.set("bgm_progression", 1)` inside map event.

Reply:
Can you provide your banks?

Reply:
And anything else that may be relevant

Reply:
I can't seem to recreate this issue

Reply:
Ah, it's because you said to use `Audio` instead of `FMODPlayer`

Reply:
Looks like this is the source of the segfault
![image](https://user-images.githubusercontent.com/54482069/178205751-7190702f-464d-4c6e-bcc8-e162dbcec4d9.png)

Reply:
> Ah, it's because you said to use Audio instead of FMODPlayer

Ah yeah, i'm sorry, i mean `FMODPlayer.set("bgm_progression", 1)`

Reply:
> Can you provide your banks?

https://drive.google.com/file/d/1QQVK3wYu0YiUNpDox7R-yIfBpA-Y-kPe/view?usp=sharing

Reply:
I have no idea why this is happening
![image](https://user-images.githubusercontent.com/54482069/178207277-61633868-c137-4fe7-b40f-53e5ea0a202b.png)

Reply:
I think a refactor of `rb_get_args` is in order

Reply:
> https://drive.google.com/file/d/1QQVK3wYu0YiUNpDox7R-yIfBpA-Y-kPe/view?usp=sharing

Thank you!

Reply:
I'm not sure how to fix this, it's on my list of things to fix later
I'll go ahead and merge this PR since it is basically done and a lot of the bindings do work fine